select.py

```diff
@@ -40,8 +40,8 @@ def add_red(students, src_name, dst_names):
 def is_valid_pair(s1, s2):
     return (s1 != s2) and not s1.paired and not s2.paired and (s1 not in s2.red) and (s2 not in s1.red)
 
-def find_match(src, candidates, max_tries = 10):
-    if candidates:
+def find_match(src, students, max_tries = 10):
```

Comment:
It is better to replace `students` which is global to a local variable, e.g `candidates`. This was the case in the previous commit. The error was actually on line 46 which still uses the global `students`. I suggest changing `students` on lines 43, 44, and 46 to `candidates`.

Reply:
Actually, `students` is not global anymore. Sorry. It previously was but I moved it inside `main`. In this case, your edits are just fine.